server/main.lua

```diff
@@ -304,54 +349,79 @@ RegisterNetEvent('qb-banking:updatePin', function(pin)
-        while bank == nil do Wait(0) end
-        while savings == nil do Wait(0) end
-        TriggerClientEvent('qb-banking:openBankScreen', src)
-        TriggerClientEvent('qb-banking:successAlert', src, Lang:t('success.savings_withdrawal', {value = tostring(amount)}))
-        TriggerEvent('qb-log:server:CreateLog', 'banking', 'Banking', 'red', "**"..GetPlayerName(xPlayer.PlayerData.source) .. " (citizenid: "..xPlayer.PlayerData.citizenid.." | id: "..xPlayer.PlayerData.source..")** made a savings withdrawal of $"..tostring(amount).." successfully.")
+    
+    local auth = validatePosition(GetEntityCoords(GetPlayerPed(src)))
+    if auth.isNearBank then
+        local currentSavings = savingsAccounts[xPlayer.PlayerData.citizenid].GetBalance()
+        if tonumber(amount) <= currentSavings then
+            local savings = savingsAccounts[xPlayer.PlayerData.citizenid].RemoveMoney(tonumber(amount), Lang:t('info.savings_to_current'))
+            local bank = xPlayer.Functions.AddMoney('bank', tonumber(amount), 'banking-quick-withdraw')
+            while bank == nil do Wait(0) end
+            while savings == nil do Wait(0) end
+            TriggerClientEvent('qb-banking:openBankScreen', src)
+            TriggerClientEvent('qb-banking:successAlert', src, Lang:t('success.savings_withdrawal', {value = tostring(amount)}))
+            TriggerEvent('qb-log:server:CreateLog', 'banking', 'Banking', 'red', "**"..GetPlayerName(xPlayer.PlayerData.source) .. " (citizenid: "..xPlayer.PlayerData.citizenid.." | id: "..xPlayer.PlayerData.source..")** made a savings withdrawal of $"..tostring(amount).." successfully.")
+        end
+    else
+        TriggerClientEvent('QBCore:Notify', src, Lang:t('error.too_far_away'), "error")
+        TriggerEvent('qb-log:server:CreateLog', 'banking', 'Banking', "red", "**"..GetPlayerName(xPlayer.PlayerData.source) .. " (citizenid: "..xPlayer.PlayerData.citizenid.." | id: "..xPlayer.PlayerData.source..")** has attempted to withdraw from savings whilst not at a banking location, Nearest bank = "..auth.nearestDistance)
     end
 end)
 
 RegisterNetEvent('qb-banking:createSavingsAccount', function()
     local src = source
     local xPlayer = QBCore.Functions.GetPlayer(src)
-    local success = createSavingsAccount(xPlayer.PlayerData.citizenid)
-    repeat Wait(0) until success ~= nil
-    TriggerClientEvent('qb-banking:openBankScreen', src)
-    TriggerClientEvent('qb-banking:successAlert', src, Lang:t('success.opened_savings'))
-    TriggerEvent('qb-log:server:CreateLog', 'banking', 'Banking', "lightgreen", "**"..GetPlayerName(xPlayer.PlayerData.source) .. " (citizenid: "..xPlayer.PlayerData.citizenid.." | id: "..xPlayer.PlayerData.source..")** opened a savings account")
+    local auth = validatePosition(GetEntityCoords(GetPlayerPed(src)))
+    if auth.isNearBank then
+        local success = createSavingsAccount(xPlayer.PlayerData.citizenid)
+        repeat Wait(0) until success ~= nil
```

Comment:
Also no repeat statement needed here, just an if statemenf as said above for the while loops